src/codegen/codegen_c_visitor.cpp

```diff
@@ -1909,9 +1896,9 @@ void CodegenCVisitor::print_eigen_linear_solver(const std::string& float_type, i
 
 std::string CodegenCVisitor::internal_method_arguments() {
     if (ion_variable_struct_required()) {
-        return "id, pnodecount, inst, ionvar, data, indexes, thread, nt, v";
+        return "id, pnodecount, inst, ionvar, data, indexes, thread, nt, ml, v";
```

Comment:
Since the `global` functions are accessible through the `inst`, do we need to also pass a pointer to `ml` as well? Maybe there is some other reason for this that I haven't understood.
Same question applies to other places were `ml` was added as well

Reply:
As mentioned in slack,

When we have to access global or any other data, we have to get via ml->instance.
In nmodl, we have Instance struct but on the mod2c side, we have _threadargsproto_namespace as args to functions:

> #define _threadargsproto_namespace int _iml, int _cntml_padded, double* _p, coreneuron::Datum* _ppvar, coreneuron::ThreadDatum* _thread, coreneuron::NrnThread* _nt, double v

In order to reach to globals, we have now changed it to

> #define _threadargsproto_namespace int _iml, int _cntml_padded, double* _p, coreneuron::Datum* _ppvar, coreneuron::ThreadDatum* _thread, coreneuron::NrnThread* _nt, coreneuron::Memb_list* _ml, double v

As this prototype change is everywhere (including coreneuron side), we have to change this.

(at least this is what I remember)